File: examples/lh-server/python/09-external-events/workflow.py
from littlehorse.workflow import Workflow, WorkflowThread

EXTERNAL_EVENT_NAME = "name-posted"


def get_workflow() -> Workflow:
    def wfLogic(wf: WorkflowThread) -> None:
        name = wf.declare_str("name")
        event_payload = wf.wait_for_event(EXTERNAL_EVENT_NAME)
        name.assign(event_payload)
        wf.execute("greet", name)

    return Workflow("external-event-example", wfLogic)

File: examples/lh-server/go/09-external-events/workflow.go
package main

import (
	"github.com/littlehorse-enterprises/littlehorse/sdk-go/lhproto"
	"github.com/littlehorse-enterprises/littlehorse/sdk-go/littlehorse"
)

const ExternalEventDefName = "name-posted"

func WfLogic(wf *littlehorse.WorkflowThread) {
	name := wf.DeclareStr("name")
	eventPayLoad := wf.WaitForEvent(ExternalEventDefName).RegisteredAs(lhproto.VariableType_STR)
	name.Assign(eventPayLoad)
	wf.Execute("greet", name)
}
